Drop require_nested and include_concern

jrafanie opened the pull request:
Zeitwerk completely removes the need for these methods. See also: https://github.com/ManageIQ/manageiq/pull/22854

Fryguy commented on app/models/manageiq/providers/ovirt/inventory/collector.rb:
```diff
@@ -1,8 +1,5 @@
 class ManageIQ::Providers::Ovirt::Inventory::Collector < ManageIQ::Providers::Inventory::Collector
   # TODO: review the changes here and find common parts with ManageIQ::Providers::Ovirt::InfraManager::Inventory::Strategies::V4
```
This comment was attached to those require lines... Not sure if it should move somewhere else

jrafanie replied:
We have similar/copied lines here: https://github.com/ManageIQ/manageiq-providers-red_hat_virtualization/pull/40/files#diff-f2f5d4e887f15b36afacf467c592be51b1b1055f4aa04d5c553f77042620c671R2

Not sure what it's referring to but I can come back to it when I finish up the other PRs.  Maybe it was prior code that was removed and left orphaned. https://twitter.com/nzkoz/status/538892801941848064

agrare commented:
@jrafanie do these PRs have any dependencies?  On https://github.com/ManageIQ/manageiq/pull/22854 or anything else?

jrafanie commented:
> @jrafanie do these PRs have any dependencies? On [ManageIQ/manageiq#22854](https://github.com/ManageIQ/manageiq/pull/22854) or anything else?

@agrare We've been running with zeitwerk by default since https://github.com/ManageIQ/manageiq/pull/22488, so that's the dependency.

Technically, here is where we dropped support for the classic autoloader (merged about 1 month ago), making usage of these methods no longer needed: https://github.com/ManageIQ/manageiq/pull/22801

jrafanie commented:
> > @jrafanie do these PRs have any dependencies? On [ManageIQ/manageiq#22854](https://github.com/ManageIQ/manageiq/pull/22854) or anything else?
> 
> @agrare We've been running with zeitwerk by default since [ManageIQ/manageiq#22488](https://github.com/ManageIQ/manageiq/pull/22488), so that's the dependency.
> 
> Technically, here is where we dropped support for the classic autoloader (merged about 1 month ago), making usage of these methods no longer needed: [ManageIQ/manageiq#22801](https://github.com/ManageIQ/manageiq/pull/22801)

Updated https://github.com/ManageIQ/manageiq/pull/22854 with this information.
